src/mavis/align.py

```diff
@@ -17,6 +17,9 @@
 from .types import ReferenceGenome
 from .util import logger
 
+if TYPE_CHECKING:
+    from .bam.cache import BamCache
```

Comment:
Any particular reason this annotation would be optional?

Reply:
This is just to avoid adding any circular imports